### PR TITLE
A few modernizations for the general documentation, and the Java examples

### PR DESCRIPTION
**Pull request by amotl**
Hi there,

first things first, I wish all of you a happy new year.

This is just a little maintenance patch, in order to improve the documentation, the repository layout, and to make the Java examples build again. Other than this, it will also address GHSA-4jrv-ppp4-jm57 at the `java-spark` example.

Based on a review item by @proddata, work on this has also spawned [Provisioning CrateDB in Docker sandbox](https://community.crate.io/t/provisioning-cratedb-in-docker-sandbox/1326). Thanks!

With kind regards,
Andreas.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -11,33 +13,51 @@ each using a different `client library`_ to communicate with CrateDB_.
 Prerequisites
 =============
 
-You need install and run CrateDB 0.54.1 or higher before you set things up.
+- You need install and run CrateDB 0.54.1 or higher before you set things up.
```
Comment (proddata) [inline README.rst:16]: Can we change that to the latest or 5.0.x 🤷 
We don't really support older versions anyway

File: README.rst [hunk 2]
```diff
@@ -11,33 +13,51 @@ each using a different `client library`_ to communicate with CrateDB_.
...
-Setup
-=====
+If you choose to use Docker for running CrateDB on your workstation, those
+commands outline the process::
 
-Once CrateDB instance is running, create the required schema and import the
-country data:
+    # Define CrateDB version.
+    export CRATEDB_VERSION=5.1.2
+    export CRATEDB_IMAGE=crate:${CRATEDB_VERSION}
```
Comment (proddata) [inline README.rst:36]: Is there a good reason not to use `latest` ❓

File: frontend/README.rst [hunk 3]
```diff
@@ -1,26 +1,28 @@
+.. highlight:: sh
+
 ========
 Frontend
 ========
 
-The frontend is written in AngularJS_ 1.4.8.
+- The frontend is written in `AngularJS`_ 1.4.8.
```
Comment (proddata) [inline frontend/README.rst:7]: 🙈 🙈 🙈 🙈 🙈 🙈 🙈

File: README.rst (hunk 1 above)
Comment (amotl) [reply README.rst]: Thanks. Fixed with 79278325.

File: README.rst (hunk 2 above)
Comment (amotl) [reply README.rst:36]: No. Also fixed with 79278325.

File: README.rst [hunk 4]
```diff
@@ -11,33 +13,51 @@ each using a different `client library`_ to communicate with CrateDB_.
...
 
-.. code-block:: sh
+    # Start CrateDB.
+    docker run -it --rm \
+        --volume=$(pwd)/sql:/sql \
+        --publish=4200:4200 --publish=5432:5432 ${CRATEDB_IMAGE}
+
+    # Populate schema and data.
+    docker run --rm -it --network=host --volume=$(pwd)/sql:/sql ${CRATEDB_IMAGE} \
+        sh -c 'cat /sql/schemas.sql | crash'
```
[the remark this answers is not in the document]
Comment (amotl) [reply README.rst]: Thanks. I've improved this section with 40a2550.

**Review by proddata — APPROVED** [review]

File: java-spring/documentation.md [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
 
 **Note:** The application is written in Java 11.
 
 ## Installation
 
-In this sample application, we use the [Crate JDBC driver 2.6.0][3] which
-uses [PostgreSQL wire protocol v3][1]. To obtain the [Crate JDBC driver][4],
+In this sample application, we use the [CrateDB JDBC driver 2.6.0][3], which
+uses [PostgreSQL wire protocol v3][1]. To obtain the [CrateDB JDBC driver][4],
 please follow [instructions][5] for the build tool of your choice.
```
[the remark this answers is not in the document]
Comment (amotl) [reply java-spring/documentation.md:9]: Better change the link to https://mvnrepository.com/artifact/io.crate/crate-jdbc/2.6.0 ?

File: java-spark/documentation.md [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
-Staring from 0.57.0 Crate supports for the [PostgreSQL wire protocol v3][1].
+Staring from 0.57.0, CrateDB supports for the [PostgreSQL wire protocol v3][1].
 To get more detail information on the protocol support by Crate,
 please have a look at the [documentation][2].
 
-In the sample application, we use the [Crate JDBC driver 2.0.x][3] which
-uses [PostgreSQL wire protocol v3][1]. To obtain the [Crate JDBC driver][4],
+In the sample application, we use the [CrateDB JDBC driver 2.0.x][3] which
+uses [PostgreSQL wire protocol v3][1]. To obtain the [CrateDB JDBC driver][4],
 please follow [instructions][5] for the build tool of your choice.
```
[the remark this answers is not in the document]
Comment (amotl) [reply java-spark/documentation.md:11]: Thanks for spotting. I will update the links.

File: java-spring/README.rst [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+A Java backend app using `Spring Data JDBC`_ , `Spring Boot`_ and the
+`CrateDB JDBC driver`_ contributed by Jürgen Mayrbäurl <jurgenma@microsoft.com>.
+
+
+Prerequisites
+=============
+
+- Make sure `Java 11`_ installed.
+- CrateDB 4.1.4 with CrateDB JDBC driver 2.6.0 was used for testing.
```
[the remark this answers is not in the document]
Comment (amotl) [reply java-spring/README.rst:15]: Yeah, but this is a statement by the original author. I did not verify functionality of the applications yet.

File: java-spring/documentation.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (amotl) [reply java-spring/documentation.md:9]: Fixed with 3b24f56d31.

File: java-spark/documentation.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (amotl) [reply java-spark/documentation.md:11]: Fixed with 3b24f56d31.

**Pull request merged**